### DISCUSSION TITLE
Using Content Block while Ublock is OFF?

### DISCUSSION THREAD

**Discussion by nokaoii** [uBlockOrigin/uBlock-issues#1788] [Ideas] [answered]
Situation:

1. Visit site XYZ.site
2. The site detects Ublock is enabled and suggest you disable it OR forces you to do it.
3. You disable it and site XYZ.site loads normally with ads
4. You realize, site XYZ.site is decent enough to show you ads to help them make money BUT there's one annoying page element/ad that you want to remove, leaving all the others.

Currently there's no way to block HTML content unless uBlock is fully enable, thus triggering the "you're running an adblocker".

Could there be a future option where uBlock can be full OFF, but still enable the content blocking function (since it's probably just a CSS display flip), so some certain elements can be visually be blocked? Obviously the ad content would still load even if blocked with CSS, but visually the site would be improved and we could support the site via their ads.

**Comment by gorhill** [comment]
uBO defuses anti-blocker by default, if you find such instance not handled by uBO, it's a filter issue, to be reported to filter list maintainers -- and also, nobody can help you if you do not provide any specific information where you suffer an issue, `xyz.site` is not an actual valid site, it's unactionable.

See [filter lists issue tracker](https://github.com/uBlockOrigin/uAssets/issues?q=is%3Aissue+is%3Aclosed+Anti-adblock), anti-blockers as reported by users are regularly worked around.

**Comment by nokaoii** [comment]
Thank you for editing your response to be a little less rude.

I'm happy to clarify.

You go to any random site, and decide you like their content so much (or whatever) that you disable uBLOCK on their site. BUT that site has ONE annoying ad or HTML element that you could nuke with uBLOCK's element picker. The element picker is not available when UBLOCK is in the "OFF" mode for the site. I would like to see the element picker be enabled even for sites where uBLOCK is off.

I don't want the ad blocked, loading it is fine, I want to support whatever the random site that I like. I just want to nuke a visual section of annoying content on their site.

Thank you for your time.

**Reply by gorhill** [reply, marked as answer]
> Thank you for editing your response to be a little less rude.

Yes sorry about this, I have confused post in Discussions as post in the issue tracker a couple of times now, I need to be more careful about this. Typically I do not have much tolerance for posts in issue tracker which bypass the template and basic steps and I erroneously though I was dealing with one of these.

This has been requested in the past (on Reddit if I remember correctly) and I declined, and still feel the same, I consider this beyond the primary purpose of uBO, a content blocker which can be easily turned off.

I suggest you use [Stylus](https://github.com/openstyles/stylus/) for this, that matches its primary purpose.

**Comment by nokaoii** [comment]
I can imagine how annoying that would be.
Thank you for letting me know the decision and even the option for Stylus. :)

Also, thank you for the great Extension. I can't imagine the web without it.
